examples/basic_request_components/src/main.rs

```diff
@@ -0,0 +1,66 @@
+//! A basic example showing the request components
+
+extern crate gotham;
+extern crate hyper;
+extern crate mime;
+
+use hyper::{Body, Headers, HttpVersion, Method, Response, StatusCode, Uri};
+
+use gotham::http::response::create_response;
+use gotham::state::{FromState, State};
+use gotham::router::Router;
+use gotham::router::builder::{build_simple_router, DefineSingleRoute, DrawRoutes};
+
+/// Show the request components by printing them.
+pub fn show_request(state: State) -> (State, Response) {
+    {
+        let method = Method::borrow_from(&state);
+        let uri = Uri::borrow_from(&state);
+        let http_version = HttpVersion::borrow_from(&state);
+        let headers = Headers::borrow_from(&state);
+        let body = Body::borrow_from(&state);
+        println!("Method: {:?}", method);
+        println!("URI: {:?}", uri);
+        println!("HTTP Version: {:?}", http_version);
+        println!("Headers: {:?}", headers);
+        println!("Body: {:?}", body);
+    }
+    let res = create_response(&state, StatusCode::Ok, Some((vec![], mime::TEXT_PLAIN)));
+
+    (state, res)
+}
+
+/// Create a `Router`
+fn router() -> Router {
+    build_simple_router(|route| {
+        route.get("/").to(show_request);
```

Comment:
It might be worth using [`associations`](https://github.com/gotham-rs/gotham/blob/further-examples-structure-refinements/examples/routing/associations/src/main.rs#L48) here to support `get` and `post`.

Once that is done it might be feasible to extend the example handler to grab body content when present and dump it to stdout also, as currently implemented I'm not immediately sure what the output would actually be if body content is present in the request.